samples/lib/widgets/pagination.dart

```diff
@@ -0,0 +1,176 @@
+import 'package:flutter/material.dart';
+
+import 'package:auto_route/auto_route.dart';
+import 'package:forui/forui.dart';
+
+import 'package:forui_samples/sample.dart';
+
+@RoutePage()
+class PaginationPage extends Sample {
+  final FPaginationController controller;
+
+  PaginationPage({@queryParam super.theme, @queryParam super.maxWidth = 400})
+    : controller = FPaginationController(pages: 10);
+
+  @override
+  Widget sample(BuildContext context) =>
+      Column(mainAxisAlignment: MainAxisAlignment.center, children: [FPagination(controller: controller)]);
+}
+
+@RoutePage()
+class PaginationSiblingsPage extends Sample {
+  final FPaginationController controller;
+
+  PaginationSiblingsPage({@queryParam super.theme, @queryParam super.maxWidth = 500})
+    : controller = FPaginationController(pages: 20, siblings: 2, initialPage: 9);
+
+  @override
+  Widget sample(BuildContext context) =>
+      Column(mainAxisAlignment: MainAxisAlignment.center, children: [FPagination(controller: controller)]);
+}
+
+@RoutePage()
+class PaginationShowEdgesPage extends Sample {
+  final FPaginationController controller;
+
+  PaginationShowEdgesPage({@queryParam super.theme}) : controller = FPaginationController(pages: 8, showEdges: false);
+
+  @override
+  Widget sample(BuildContext context) =>
+      Column(mainAxisAlignment: MainAxisAlignment.center, children: [FPagination(controller: controller)]);
+}
```

Comment:
All these can be combined into a single widget with different query params. See badge for an example.